proposal/csvt32745/proposal.rst

```diff
@@ -0,0 +1,183 @@
+
+.. Analyze how your system takes input, produces results, and performs any other
+.. operations.
+
+.. Describe the system's work flow.  You may consider to use a flow chart but it
+.. is not required.
+
+.. Specify the constraints assume in your system.  Describe how it is modularized.
+
+Initialization
+    Unstructured list of raw animation (motion capture animation prefered), 
+    which is preferably marked with specific label (like Running, Jumping) in Godot editor by the creator.
+
+Input (Runtime)
+    Specific label, Velocities and Positions of the avatar
+
+Output
+    The fluid motion of avatar depends on physic parameters and the player's input.
+
+
+API description
+===============
+
+.. Show how a user programmatically uses your system.  You are supposed to
+.. implement the system using both C++ and Python.  Describe how a user writes a
+.. script in the system.  If you provide API in both C++ and Python, describe
+.. both.
+
+
+I will wrap my C++ module into Godot engine editor, and programmers are still allowed to call the plugin function by themselves.
+
+Compiling
+    | Though the Godot engine executable is released directly, it needs to be re-compiled when binding other C++ modules.
+    | And we need **SCons** to build the program according to `official instruction <https://docs.godotengine.org/en/latest/development/compiling/introduction_to_the_buildsystem.html>`__.
+    | The compilation part of this repository will include
+    
+    - Source codes of C++ modules
+    - Importing-script of engine
+    - A SCons scipt 
+
+Usage
+    The usage of plugin is the same in C++ or scipting in engine.
+    And the following C++ classes are going to be used in in-editor script.
+
+    - `Class Pose`
+        - Attributes used to compute the cost.
+        - Velocity, position of some bones
+        - Specific Labels
+
+    - `Class AnimationMotionMatching`
+        - The major working component of motion Matching
+        - Attached on a avatar
+        - Member
+            - `vector<Pose> m_Pose` : Animation database
+            - `AnimationPlayer m_AnimPlayer` : Animation player of the avatar
+            - `To be defined...`
+        - Method
+            - `void AnimUpdate()` : Select and update the next animation
+            - `float ComputeCost(const Pose& currentPose, const Pose& candidatePose)` : Cost Function
+            - `To be defined...`
+    
+
+
+Engineering infrastructure
+==========================
+
+.. Describe how you plan to put together the build system, testing framework, and
+.. documentation.  Show how you will do version control.
+
+.. You may use continuous integration, but it is not required.  If you use it,
+.. describe how it works in your code development.
+
+Architecture
+    | I will implement Motion Matching as a C++ plugin in `Godot <https://godotengine.org/>`_ Engine,
+    | Though scipts in Godot are written in High-Level language like C# and its own language - gdscript,

```

Comment:
The term project expects you to exercise the hybrid architecture of the high-/low-level C++ and scripting language.  You should pick one, if not two, of C# or gdscript and exhibit your capability of working with the hybrid architecture.